userspace/libsinsp/container.cpp

```diff
@@ -704,11 +704,29 @@ bool sinsp_container_manager::parse_docker(sinsp_container_info* container)
 	const Json::Value& net_obj = root["NetworkSettings"];
 
 	string ip = net_obj["IPAddress"].asString();
-	if(inet_pton(AF_INET, ip.c_str(), &container->m_container_ip) == -1)
+	if(ip.empty())
 	{
-		ASSERT(false);
+		const Json::Value& hconfig_obj = root["HostConfig"];
+		string net_mode = hconfig_obj["NetworkMode"].asString();
+		if(strncmp(net_mode.c_str(), "container:", strlen("container:")) == 0)
+		{
+			sinsp_container_info pcnt;
+			pcnt.m_id = net_mode.substr(net_mode.find(":") + 1);
```

Comment:
Looking at the documentation, it seems like `container:` for specifying the network can also point to a container name rather than a container id, which is a case we wouldn't support. Did you check by any chance that Kubernetes always uses the id?

Reply:
I manually verified with docker and you're right, the network mode can also contains names. On the other hand, `GET /containers/<container name>/json` works as well. I can refactor it a little bit to avoid the calls at `get_container`/`parse_docker` and just do the request, and that should work for both names and IDs. What do you think?

Reply:
I think that as long as we support all Kubernetes configuration and we don't risk running into infinite recursion, it's fine to ignore the problem.

Reply:
Verified that our code support both container ID and container name.